### PR TITLE
feat: add tests details and make it reachable

### PR DESCRIPTION
**Pull request by WilsonNet**
Description:

Add test details page, and update the test tabs for the demo so it show all tests instead of groups 

- Add test details endpoint
- Add test details page
- Implemented `TestsByCommitHash` view to handle the new endpoint.
- Updated `testDetailsView.py` to include status count in response.
- Created new request script `tests-by-commit-hash.sh`.
- Added new API hooks and components in the dashboard for the new endpoint.
- Updated localization messages and routes for the new feature.

## How to test

Go to a revision that has a lot of tests
go to the tests tab
and click any test

### CODE REVIEW

File: dashboard/src/pages/TestDetails/TestDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    0,
+  );
+  const chartElements = [
+    {
+      label: <FormattedMessage id="bootsTab.success" />,
+      value: statusCount['PASS'] ?? 0,
+      color: Colors.Green,
+    },
+    {
+      label: <div>Fail</div>,
```
Comment (mari1912) [inline dashboard/src/pages/TestDetails/TestDetails.tsx:68]: use intl

File: dashboard/src/pages/TestDetails/TestDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      value: statusCount['FAIL'] ?? 0,
+      color: Colors.Red,
+    },
+    {
+      label: <div>Skip</div>,
```
Comment (mari1912) [inline dashboard/src/pages/TestDetails/TestDetails.tsx:73]: ditto

File: dashboard/src/pages/TestDetails/TestDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+      value: statusCount['MISS'] ?? 0,
+      color: Colors.Yellow,
+    },
+    {
+      label: <div>Done</div>,
+      value: statusCount['DONE'] ?? 0,
+      color: Colors.Yellow,
+    },
+    {
+      label: <div>Done</div>,
```
Comment (mari1912) [inline dashboard/src/pages/TestDetails/TestDetails.tsx:88]: error instead of done

File: dashboard/src/pages/TestDetails/TestDetails.tsx (hunk 2 above)
Comment (mari1912) [reply dashboard/src/pages/TestDetails/TestDetails.tsx:73]: for all the values in the chart element

File: dashboard/src/pages/TestDetails/TestDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+      value: statusCount['FAIL'] ?? 0,
+      color: Colors.Red,
+    },
+    {
+      label: <div>Skip</div>,
+      value: statusCount['SKIP'] ?? 0,
+      color: Colors.Yellow,
+    },
+    {
+      label: <div>Skip</div>,
```
Comment (mari1912) [inline dashboard/src/pages/TestDetails/TestDetails.tsx:78]: miss instead of skip

File: dashboard/src/pages/TestDetails/TestDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+  return (
+    <BaseCard
+      title={title}
+      content={
+        <StatusChartMemoized
+          type="chart"
+          elements={filteredChartElements}
+          pieCentralLabel="Statuses"
+          pieCentralDescription={<>{totalCount}</>}
```
Comment (mari1912) [inline dashboard/src/pages/TestDetails/TestDetails.tsx:106]: nit: we should use intl in the future

File: dashboard/src/pages/TestDetails/TestDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+      <pre className="overflow-x-auto">
+        {JSON.stringify(currentTest, null, 2)}
+      </pre>
+      */}
+
+      <div className="w-100 px-5">
+        <Breadcrumb className="pb-6 pt-6">
+          <BreadcrumbList>
+            <BreadcrumbItem>
+              <BreadcrumbLink to="/tree">Revisions</BreadcrumbLink>
```
Comment (mari1912) [inline dashboard/src/pages/TestDetails/TestDetails.tsx:239]: ditto

File: dashboard/src/pages/TestDetails/TestDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+            </BreadcrumbItem>
+            <BreadcrumbSeparator />
+            <BreadcrumbLink to={`/tree/$treeId`} params={{ treeId: treeId }}>
+              Tree Details
+            </BreadcrumbLink>
+            <BreadcrumbSeparator />
+            <BreadcrumbItem>
+              <BreadcrumbPage>Test Details</BreadcrumbPage>
```
Comment (mari1912) [inline dashboard/src/pages/TestDetails/TestDetails.tsx:247]: same

File: dashboard/src/pages/TreeDetails/Tabs/Tests/ImprovisedTestsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+import { useTestsByCommitHash } from '@/api/TreeDetails';
+
+const headerLabelIds: string[] = [
+  'test.id',
+  'test.status',
+  'test.path',
+  'test.compiler',
+  'test.architecture',
+];
```
Comment (mari1912) [inline dashboard/src/pages/TreeDetails/Tabs/Tests/ImprovisedTestsTable.tsx:19]: we should show the labels instead of this

**Review by mari1912 — APPROVED** [review]

**Comment by WilsonNet** [comment]
Let's merge as this for the demo and fix those minor issues later on, keep this PR for tracking

**Pull request merged**